src/vsr/journal.zig

```diff
@@ -586,6 +586,20 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
             return copied;
         }
 
+        /// Returns the op of the last gap or hash-break in the headers.
+        /// Returns null when the hash chain is valid all the way back to op=0.
+        pub fn find_latest_headers_break_before(journal: *const Journal, op_max: u64) ?u64 {
+            assert(journal.status == .recovered);
+            assert(journal.header_with_op(op_max) != null);
+
+            var op: u64 = op_max;
+            while (op > 0) : (op -= 1) {
+                const header_next = journal.header_with_op(op).?;
+                const header_prev = journal.header_with_op(op - 1) orelse return null;
+                if (header_prev.checksum != header_next.parent) return null;
+            } else return null;
+        }
+
```

Comment:
Does this always return null?

Reply:
Good catch.. No, that's a bug. Though now I'm wondering why it made the test pass :thinking:

Reply:
Maybe the first two `return null`s should be `return op-1`?

Reply:
https://github.com/tigerbeetledb/tigerbeetle/issues/369#issuecomment-1360392012 :D

Reply:
> Maybe the first two return nulls should be return op-1?

Yeah, I think so too, but I need to figure out why it passes the tests without that

Reply:
Aha, I didn't check the final fix against the correct seed. The `return null` fails as expected after all, and `return op - 1` fixes it.